test/test-python-shell.ts

```diff
@@ -225,6 +225,22 @@ describe('PythonShell', function () {
                 scriptPath: pythonFolder
             };
         })
+
+        it('should run PythonShell normally without access to std streams', function (done) {
+            var pyshell = PythonShell.run('exit-code.py', {
+                // 3 different ways of assigning values to the std streams in child_process.spawn()
+                // * ignore - pipe to /dev/null
+                // * inherit - inherit fd from parent process;
+                // * process.stderr - pass output directly to that stream.
+                stdio: ['ignore', 'inherit', process.stderr],
+                args: ["0"]
+            }, done);
+
+            should(pyshell.stdin).be.eql(null);
+            should(pyshell.stdout).be.eql(null);
+            should(pyshell.stderr).be.eql(null);
```

Comment:
Why is stdout and stderr null? Because these both inherit the parent's stdout / stderr shouldn't they have a value?

Reply:
Yes. The underlying code just does a file descriptor dup() so the child process just inherits the file descriptors directly and then all output goes to /dev/tty (for instance, if you are working in a terminal) bypassing any streams in memory in the local process.